src/compute-types/src/dataflows.rs

```diff
@@ -108,6 +114,33 @@ impl<T> DataflowDescription<Plan<T>, (), mz_repr::Timestamp> {
         as_of.try_step_forward().as_ref() == until.as_option()
     }
 
+    /// TODO
```

Comment:
This function is load-bearing, but I have a hard time understanding what it's supposed to do. Please add documentation and try to clarify the implementation. For example, why are the or-conditions at the top `..is_none() || (A || B) || !C`? The parens are redundant, right?